pkg/util/supervisor.go

```diff
@@ -34,3 +40,58 @@ func IsSupervisorType(input interface{}) (bool, error) {
 		return false, fmt.Errorf("unexpected type %s", reflect.TypeOf(input))
 	}
 }
+
+// SetControllerReferenceWithOverride sets owner as a Controller OwnerReference on controlled.
+// This is used for garbage collection of the controlled object and for
+// reconciling the owner object on changes to controlled (with a Watch + EnqueueRequestForOwner).
+// Since only one OwnerReference can be a controller, it returns an error if
+// there is another OwnerReference with Controller flag set unless it was a legacy controller owner.
+func SetControllerReferenceWithOverride(owner, controlled metav1.Object, scheme *runtime.Scheme) error {
+	// Validate the owner.
+	ro, ok := owner.(runtime.Object)
+	if !ok {
+		return fmt.Errorf("%T is not a runtime.Object, cannot call SetControllerReference", owner)
+	}
+
+	// Create a new controller ref.
+	gvk, err := apiutil.GVKForObject(ro, scheme)
+	if err != nil {
+		return err
+	}
+	ref := metav1.OwnerReference{
+		APIVersion: gvk.GroupVersion().String(),
+		Kind:       gvk.Kind,
+		Name:       owner.GetName(),
+	}
+
+	deleteAllControllers(controlled, ref)
+
+	return controllerutil.SetControllerReference(owner, controlled, scheme)
+}
+
+// deleteAllControllers Removes existing controller reference from controlled object.
+func deleteAllControllers(controlled metav1.Object, ref metav1.OwnerReference) {
+	owners := controlled.GetOwnerReferences()
+	for i := range owners {
+		if owners[i].Controller != nil && *owners[i].Controller && !referSameObject(owners[i], ref) {
```

Comment:
`!referSameObject(owners[i], ref)` is this check for the follow up reconcile loops for the same object? Can you add a comment about the same if that's the case.